src/ulid.cr

```diff
@@ -19,6 +19,23 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # is ulid the correct size AND only contain valid characters?
+    ulid.size == TIME_LEN + RANDOM_LEN && (ulid =~ /[^0123456789ABCDEFGHJKMNPQRSTVWXYZ]/).nil?
+  end
+
+  # TODO #valid!(ulid : String) => returns nil, raise appropriate errors in invalid
+
+  def seed_time(ulid : String) : Time
+    Time.unix_ms(ulid[0..9].to_i64(32))
+  end
+
```

Comment:
A nice addition to make you be `ULID.parse(String) : { Time, String }`. This can check the format is valid (raising an exception if not), the parse out the time and random components.

Reply:
Started an initial attempt. Have moved that work in progress over to another branch to PR later.